### PR TITLE
Update : Aprove&RejectTasksNotification

### PR DESCRIPTION
**Pull request by Yousif-Alsheikh**

### CODE REVIEW

**Review by KAJOiq — APPROVED** [review]

File: Contexts/MainAppContext.cs [hunk 1]
```diff
@@ -143,6 +143,28 @@ protected override void OnModelCreating(ModelBuilder builder)
                                                        .HasForeignKey(sn => sn.FreelancerId)
                                                        .HasPrincipalKey(u => u.Id)
                                                        .OnDelete(DeleteBehavior.NoAction);
+            /////
+            builder.Entity<TaskApprovalNotification>().HasOne<TaskEntity>()
```
Comment (MohammedLilo) [inline Contexts/MainAppContext.cs:147]: I suppose that the relationship between `TaskApprovalNotification` and `TaskEntity` should be one-to-one, but NVM, we can change it later if necessary

File: Controllers/Mobile/v1/ProfileController.cs [hunk 2]
```diff
@@ -189,13 +190,33 @@ public async Task<IActionResult> DeleteProfilePictureAsync()
...
+
+        [HttpGet("client-activity/{clientId}")]
+        public async Task<IActionResult> GetClientActivityByIdAsync( [FromRoute] long clientId,
+            [FromQuery] int page = 0,
+            [FromQuery] int pageSize = Constants.CLIENT_ACTIVITY_DEFAULT_PAGE_SIZE)
+        {
+            if (!ModelState.IsValid)
+                return base.CustomBadRequest();
+
+            Client? StoredId =await profileService.FindClientAsync(clientId);
```
Comment (MohammedLilo) [inline Controllers/Mobile/v1/ProfileController.cs:202]: nigga change `StoredId` to `storedClient`

File: Controllers/Mobile/v1/ProfileController.cs [hunk 3]
```diff
@@ -189,13 +190,33 @@ public async Task<IActionResult> DeleteProfilePictureAsync()
...
+            if (StoredId == null)
+                return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "Client not found."));
+            PaginatedResult<Project> paginatedProjects = await profileService.FindClientActivitiesAsync(clientId, page, pageSize);
+            if (paginatedProjects == null)
```
Comment (MohammedLilo) [inline Controllers/Mobile/v1/ProfileController.cs:206]: projects will not be null if the client has no projects, instead you will get an empty list

File: Models/DTOs/ClientActivityOutputDTO.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            Title = project.Title;
+            Description = project.Description;
+            CreationTime= StringOperations.GetTimeAgo( project.CreatedAt);
+            TotalLikes = project.ProjectLikes.Count;
+            TotalComment = project.Comments.Count;
+            Duration = project.Duration;
+            DurationString = StringOperations.ConvertDaysToMonthsAndYears(project.Duration);
+            Budget = project.Budget;
+        }
+        public static ClientActivityOutputDTO FromClientActivity(Project project) => new ClientActivityOutputDTO(project);
```
Comment (MohammedLilo) [inline Models/DTOs/ClientActivityOutputDTO.cs:30]: this method should be renamed to `FromProject`

File: Services/ProfileService.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            var query = mainAppContext.Projects.AsNoTracking().Where(p => p.ClientId == clientId)       
+                                                              .Include(p => p.ProjectLikes)
+                                                              .Include(p=>p.Comments)
+                                                              .AsQueryable();
+
+            List<Project>? storedProjects = await query.OrderByDescending(p => p.CreatedAt)
+                                                       .Skip(pageNumber * pageSize)
+                                                       .Take(pageSize)
+                                                       .ToListAsync();
+            return new PaginatedResult<Project>(storedProjects.Count, storedProjects);
```
Comment (MohammedLilo) [inline Services/ProfileService.cs:26]: you should put the total count of projects of this client, not the count of projects in this returned page

File: Utilities/StringOperations.cs
Comment (MohammedLilo) [inline Utilities/StringOperations.cs:1]: TLDR

**Review by MohammedLilo — COMMENTED** [review]
Just update according to the comments.
To be tested and approved.

**Review by MohammedLilo — APPROVED** [review]

**Review by MohammedLilo — APPROVED** [review]

**Pull request merged**